src/kitovu/sync/settings.py

```diff
@@ -0,0 +1,97 @@
+"""A collection of all settings wrappers and factories for the sync."""
+
+import pathlib
+import typing
+
+import yaml
+import attr
+
+from kitovu import utils
+
+
+SimpleDict = typing.Dict[str, typing.Any]
+
+
+@attr.s
+class PluginSettings:
+    """A class representing the settings of a single plugin"""
+
+    plugin_type: str = attr.ib()
```

Comment:
I find `plugin_type` a confusing name, as it's actually the *name* of a plugin (and not some kind of e.g. Python type).

However, that'd be confusing in a different way - I'll explain in a comment further down in the example config file.